### ISSUE TITLE
Test coverage

### ISSUE THREAD

**Issue by ehhc** [BoostIO/Boostnote#1826]
There is a lot of untested code. During my fixes for #1723 and #1825 i found a really huge bunch of files that are not tested at all. E.g. MarkdownPreview. Because of this a contribution is really hard because it's nearly impossible to figure out if something was broke by accident.

**Comment by Rokt33r** [comment]
You're right. We should add lots of tests. But, my time is quite limited to contribute this project. Hmm.... do you have any idea how we can distribute the tasks to other contributors?

**Comment by ehhc** [comment]
@Rokt33r  unfortunately currently I don't have any idea how to do so. It's hard to write tests if the code was created by someone else. Therefore it's hard to imagine other people who don't understand the code totaly write tests later on. I think the only practical solution is to look for a good test coverage when accepting new code (even though that might be impossible for fixes because nobody will/is able to write completely new test if there are none to use as a entry point).
Furthermore, i suggest taking a good test coverage as a main focus in new projects..

**Comment by amedora** [comment]
@Rokt33r you said [you prefer Jest ](https://github.com/BoostIO/Boostnote/issues/1707#issuecomment-374812747), but we have two test frameworks now (Jest & ava). Any ideas about we should migrate one to another or keep 2 of them as is?

**Comment by amedora** [comment]
just note, yes, the test coverage we have now is not good...

_jest --coverage_
![coverage-jest](https://user-images.githubusercontent.com/32722363/62586432-9011da00-b8f8-11e9-82ea-85d5a263a60b.png)

_nyc ava_
![coverage-nyc-ava](https://user-images.githubusercontent.com/32722363/62586438-9902ab80-b8f8-11e9-9798-cd8a87a93db6.png)

**Comment by Rokt33r** [comment]
@amedora We should migrate every test to Jest!!
